pkg/controllers/nodeclass/status/securitygroup.go

```diff
@@ -29,11 +29,11 @@ import (
 )
 
 type SecurityGroup struct {
-	securityGroupProvider securitygroup.Provider
+	SecurityGroupProvider securitygroup.Provider
 }
 
 func (sg *SecurityGroup) Reconcile(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) (reconcile.Result, error) {
-	securityGroups, err := sg.securityGroupProvider.List(ctx, nodeClass)
+	securityGroups, err := sg.SecurityGroupProvider.UpdateSecurityGroups(ctx, nodeClass)
```

Comment:
Why change the name? This isn't updating security groups; this is still Listing the security groups like before. I think the handling is the same.